explore.ts

```diff
@@ -0,0 +1,58 @@
+import { Directions } from "./contants"
+import Plateau from "./plateau";
+import validate from "./validate";
+
+const explore = (currentX: number, currentY: number, currentDirection: Directions, instruction: string, plateau: Plateau) => {
+    let x = currentX;
+    let y = currentY;
+    const compass = [Directions.N, Directions.E, Directions.S, Directions.W]
+    let currentIndex = compass.indexOf(currentDirection);
+    let direction: Directions = currentDirection;
+    const instructions = [...instruction]
+    instructions.forEach((currentInstruction) => {
+        if(currentInstruction === 'L') {
+            if(currentIndex === 0) {
+                direction = compass[compass.length-1]
+                currentIndex = compass.length-1
+            }
+            else{
+                direction = compass[currentIndex-1]
+                currentIndex = currentIndex - 1
+            }
+        }
+        else if(currentInstruction === 'R') {
+            if(currentIndex === (compass.length - 1)) {
+                direction = compass[0]
+                currentIndex = 0
+            }
+            else {
+                direction = compass[currentIndex+1]
+                currentIndex = currentIndex + 1
+            }
+        }
+        else {
+            if(direction === 'N' ) {
+                y = y + 1
+            }
+            else if(direction === 'S') {
```

Comment:
Do we have a test for this?

Reply:
Our test should cover all if else since our instruction covers all directions. Our test shouldn't be aware of underlying variables if I'm not wrong, it simply expects correct output with given input.

I'll take one more look to be sure but I don't think we should be splitting test for each if else. It is a single computation cycle that gives us desired result. These if else conditions are a part of explore logic.

Reply:
Yes, tests shouldn't be aware of the code. But if there is a piece of code that exists without a coverage, it's an indication that it was probably written without writing test first.
Let's say we write a test for "S" input, the test would exist first. Then you will fail it explicitly, and then write code to pass it. In a way, your test doesn't know about your code still, but tests each and every line.

Reply:
You're right. Only after checking test coverage I updated the instruction string. But I did follow tdd approach, just that if else block logic cannot be split into test since it's part of single logic.

I'm checking on how better I'd write or test nested if else in our function.

Reply:
Then maybe to stricter TDD. The nested block wouldn't have existed if we had written a failling test for it.

Remember - red, green, refactor :)

Reply:
I guess I should include coverage in test configuration such that test should fail if coverage isn't 100%.